containers-sig-laws/app/Main.hs

```diff
@@ -0,0 +1,102 @@
+{-# LANGUAGE TypeApplications #-}
+
+module Main where
+
+import Control.Monad
+import           Test.QuickCheck
+import qualified Map.Contrib.Laws.Int as LI
+import qualified Map.Contrib.Laws.RO.Int as LI
+import qualified Map.Contrib.Laws.Ord as LO
+import qualified Map.Contrib.Laws.RO.Ord as LO
+import qualified Map.Contrib.Laws.Hash as LH
+import qualified Map.Contrib.Laws.RO.Hash as LH
+import qualified Map.Contrib.Laws.RO.Prim as PR
+
+genSmallPairs :: (Arbitrary l, Arbitrary r) => Gen [(l, r)]
+genSmallPairs = do
+  len <- choose (0, 2)
+  replicateM len $ (,) <$> arbitrary <*> arbitrary
+
+main :: IO ()
+main = do
+  quickCheck PR.nullEmpty
+  quickCheck $ PR.nullImpliesZeroSize (genSmallPairs @Int @String)
+  quickCheck $ PR.nonZeroSizeImpliesNotNull @Int @String
```

Comment:
You don't have to this now. But I think that in our tests we don't care much about the types of keys or values, so you can have two short type aliases in this module:

```haskell
type K = Int  -- some randomly chosen type for the keys
type V = String  -- some randomly chosen type for the values
```

So later you can just write:

```haskell
quickCheck $ PR.nonZeroSizeImpliesNotNull @K @V
```

And if we want to change type of values (for example, because `String` is very slow and maybe `Text` is better) it can be done very easily.

Reply:
Having aliases is a great idea, will do it :)
One thing: I don't think changing types between `String` and `Text` will have a great impact since we don't process any of the values. But instances generated for these values by `QuickCheck` differ, so I'll take a look at what is being generated, and use better types if possible.